rust/lance/src/dataset.rs

```diff
@@ -3117,6 +3118,111 @@ mod tests {
         dataset.validate().await.unwrap();
     }
 
+    #[rstest]
+    #[tokio::test]
+    async fn test_merge_on_row_id(
+        #[values(LanceFileVersion::Legacy, LanceFileVersion::Stable)]
+        data_storage_version: LanceFileVersion,
+        #[values(false, true)] use_stable_row_id: bool,
+    ) {
+        // Tests a merge that spans multiple batches within files
+
+        // This test also tests "null filling" when merging (e.g. when keys do not match
+        // we need to insert nulls)
```

Comment:
Where is the null filling? It seems like you are providing every row id, unless I am missing something.

Reply:
> Where is the null filling? It seems like you are providing every row id, unless I am missing something.

sorry, I copy and modify another test